elastic/apm-server
# Introduce a configurable default service environment

## Motivation/summary

Introduce a configurable default service environment. This is recorded in events which have no service environment defined, and is used in agent central config when no environment is specified. There is no default default, so there is no change in behaviour unless the configuration is set.

## Checklist


## How to test these changes

1. Run apm-server configured with `apm-server.default_service_environment: abc123`
2. Send some transactions with environment set (to something else); the environment should be kept intact, and not overridden
3. Send some transactions with environment unset; the default environment should be set
4. Enable transaction aggregation, ensure all metrics have an environment associated
5. Enable central config, ensure that an agent with no environment set picks up config for the environment "abc123"

## Related issues

Closes https://github.com/elastic/apm-server/issues/4664.
